x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts

```diff
@@ -0,0 +1,114 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+
+import { i18n } from '@kbn/i18n';
+import {
+  CallClusterWithRequest,
+  DeprecationAPIResponse,
+  DeprecationInfo,
+  Request,
+} from 'src/core_plugins/elasticsearch';
+
+import { CURRENT_MAJOR_VERSION } from '../../common/version';
+
+import fakeDeprecations from './__fixtures__/fake_deprecations.json';
+
+export interface EnrichedDeprecationInfo extends DeprecationInfo {
+  index?: string;
+  node?: string;
+  actions?: Array<{
+    label: string;
+    url: string;
+  }>;
+}
+
+export interface UpgradeAssistantStatus {
+  cluster: EnrichedDeprecationInfo[];
+  nodes: EnrichedDeprecationInfo[];
+  indices: EnrichedDeprecationInfo[];
+
+  [checkupType: string]: EnrichedDeprecationInfo[];
+}
+
+export async function getUpgradeAssistantStatus(
+  callWithRequest: CallClusterWithRequest,
+  req: Request,
+  basePath: string,
+  useFakeData = false // TODO: remove
+): Promise<UpgradeAssistantStatus> {
+  let deprecations: DeprecationAPIResponse;
+
+  // Fake data for now. TODO: remove this before merging.
+  if (useFakeData) {
+    deprecations = _.cloneDeep(fakeDeprecations) as DeprecationAPIResponse;
+
+    // trigger pagination on cluster tab
+    for (let i = 0; i < 6; i++) {
+      deprecations.cluster_settings = deprecations.cluster_settings.concat(
+        deprecations.cluster_settings.map(cs => ({ ...cs, message: `${cs.message} ${i}` }))
+      );
+    }
+  } else {
+    deprecations = await callWithRequest(req, 'transport.request', {
+      path: '/_xpack/migration/deprecations',
+      method: 'GET',
+    });
+  }
+
+  return {
+    cluster: deprecations.cluster_settings,
+    nodes: deprecations.node_settings,
+    indices: getCombinedIndexInfos(deprecations, basePath),
+  };
+}
+
+// Combines the information from the migration assistance api and the deprecation api into a single array.
+// Enhances with information about which index the deprecation applies to and adds buttons for accessing the
+// reindex UI.
+const getCombinedIndexInfos = (deprecations: DeprecationAPIResponse, basePath: string) => {
+  const combinedIndexInfo: EnrichedDeprecationInfo[] = [];
+
+  Object.keys(deprecations.index_settings).forEach(indexName => {
+    deprecations.index_settings[indexName]
+      .map(d => ({ ...d, index: indexName } as EnrichedDeprecationInfo))
+      .map(d => {
+        // Add reindexing action if it's the old index deprecation warning.
+        if (d.message === `Index created before ${CURRENT_MAJOR_VERSION}.0`) {
+          d.actions = [
+            {
+              label: i18n.translate(
+                'xpack.upgradeAssistant.checkupTab.indices.reindexInConsoleButtonLabel',
+                {
+                  defaultMessage: 'Reindex in Console',
+                  description:
+                    '"Console" should be the same name used to label the app under Dev Tools -> Console',
```

Comment:
Are the actions something which we are introducing in Kibana, or does ES also expose the deprecations that way?

Reply:
Only in Kibana right now.

Reply:
Thoughts about moving this check to the UI? - my thinking is that it would drastically cut down on the payload size. Additionally, including it in the API means we have to be cognisent of API breaking changes going forward.

Reply:
note: Actions were removed from the API